routes/soldiers.js

```diff
@@ -0,0 +1,26 @@
+const soldierDb = require("../collections/soldiers");
+const url = require("url");
+
+function handleRequest(db, method, url, data, callback) {
+  switch (method) {
+    case "POST":
+      soldierDb.newSoldier(db, data, callback);
+      break;
+    case "GET":
+      const parsedUrl = new URL("http://localhost:3000/" + url);
+      if (/soldiers\/[0-9]{7}/.test(url)) {
+        soldierDb.getSoldier(db, null, url.split("/")[2], callback);
+      } else if (parsedUrl.searchParams.has("name")) {
+        soldierDb.getSoldier(db, parsedUrl.searchParams.get("name"), null, callback);
```

Comment:
do you see all these nulls in the parameters? this means it's a confusing flag parameter. refrain from having those. all paremeters should be necessary, don't ruin the method signature for comfort, it's slippery slope.